File: backend/app/domains/policy_intelligence/schemas/policy.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class PolicyRead(BaseModel):
    id: UUID
    title: str
    authority: str
    state: str | None = None
    category: str
    summary: str
    effective_date: datetime | None = None
    document_url: str | None = None


class TariffRecordRead(BaseModel):
    id: UUID
    state: str
    energy_source: str = "solar"
    tariff_type: str
    rate_per_kwh: float
    currency: str
    effective_date: datetime
    expiry_date: datetime | None = None
    source: str


class SubsidyRead(BaseModel):
    id: UUID
    name: str
    authority: str
    state: str | None = None
    amount: float | None = None
    unit: str
    status: str
    disbursement_date: datetime | None = None


class ComplianceAlertRead(BaseModel):
    id: UUID
    title: str
    authority: str
    data_source: str
    source_name: str
    source_url: str
    category: str
    summary: str | None = None
    published_at: datetime | None = None
    scraped_at: datetime
    is_active: bool = True

File: frontend/src/types/policy.ts
export interface Policy {
  id: string;
  title: string;
  authority: string;
  state: string | null;
  category: string;
  summary: string;
  effective_date: string | null;
  document_url: string | null;
}

export interface TariffRecord {
  id: string;
  state: string;
  energy_source: string;
  tariff_type: string;
  rate_per_kwh: number;
  currency: string;
  effective_date: string;
  expiry_date: string | null;
  source: string;
}

export interface Subsidy {
  id: string;
  name: string;
  authority: string;
  state: string | null;
  amount: number | null;
  unit: string;
  status: string;
  disbursement_date: string | null;
}

export interface ComplianceAlert {
  id: string;
  title: string;
  authority: string;
  data_source: string;
  source_name: string;
  source_url: string;
  category: string;
  summary: string | null;
  published_at: string | null;
  scraped_at: string;
  is_active: boolean;
}
